### PR TITLE
CQ-001 sonarqube #1

### PR DESCRIPTION
**Pull request by letier3110**
https://www.notion.so/madfissolutions/CQ-001-Code-quality-issues-discovered-by-SASTs-SonarQube-WebStorm-a1fd6c4a97cb4881b93beb16394a8a76

### CODE REVIEW

File: src/beacon/use-beacon-handler.hook.ts [hunk 1]
```diff
@@ -30,13 +30,15 @@ export const useBeaconHandler = () => {
   useEffect(() => {
     const listener = ({ url }: { url: string | null }) => beaconDeepLinkHandler(url ?? '');
 
+    let emitter: { remove: () => void } | undefined;
```
Comment (unixvb) [inline src/beacon/use-beacon-handler.hook.ts:33]: use `EmitterSubscription` type instead

File: src/styles/types.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  Baseline = 'baseline',
+  Stretch = 'stretch'
+}
+
+export enum FlexDirection {
+  Row = 'row',
+  RowReverse = 'row-reverse',
+  Column = 'column',
+  ColumnReverse = 'column-reverse'
+}
```
Comment (unixvb) [inline src/styles/types.ts:23]: yyyy, don't like this approach.  
Could we ignore sonarcube rule for these values?

File: src/components/asset-amount-input/asset-amount-input.tsx [hunk 3]
```diff
@@ -110,29 +110,34 @@ export const AssetAmountInput: FC<AssetAmountInputProps> = ({
...
         ...value,
-        amount: isDefined(inputValueRef.current)
-          ? isTokenInputType
-            ? tzToMutez(inputValueRef.current, value.asset.decimals)
-            : dollarToTokenAmount(inputValueRef.current, value.asset.decimals, exchangeRate)
-          : undefined
+        amount: getDefinedAmount(value.asset.decimals, exchangeRate)
       }),
-    [inputValueRef.current, value.asset.decimals, exchangeRate, isTokenInputType]
+    [value.asset.decimals, exchangeRate, getDefinedAmount]
```
Comment (unixvb) [inline src/components/asset-amount-input/asset-amount-input.tsx:140]: much harder to read ._.
could we increase complexity limit here?

Comment (letier3110) [reply src/components/asset-amount-input/asset-amount-input.tsx:140]: with a two-level boolean check duplication

File: src/components/collectible-icon/collectible-icon.tsx [hunk 4]
```diff
@@ -12,27 +12,38 @@ export const CollectibleIcon: FC<CollectibleIconProps> = ({ collectible, size })
...
+
+  return (
     <View
       style={{
         width: size,
         height: size,
         padding: formatSize(4)
       }}
     >
-      {isDefined(collectible.artifactUri) ? (
```
Comment (unixvb) [inline src/components/collectible-icon/collectible-icon.tsx:23]: why don't just extend this condition? 
The same but with fewer changed lines, isn't it?
image

File: src/components/activity-groups-list/activity-group-item/activity-group-type/use-activity-group-info.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import { isString } from '../../../../utils/is-string';
+import { IconNameEnum } from '../../../icon/icon-name.enum';
+
+export type IconGroupType = [IconNameEnum, string, string];
+
+export const useActivityGroupInfo = (group: ActivityGroup): IconGroupType => {
+  const colors = useColors();
+  const publicKeyHash = useSelectedAccountSelector().publicKeyHash;
+  const [iconName, iconColor, typeText] = useMemo<IconGroupType>(
+    () => handleChooseIcon(group, colors, publicKeyHash),
```
Comment (unixvb) [inline src/components/activity-groups-list/activity-group-item/activity-group-type/use-activity-group-info.ts:16]: remove `handleChooseIcon` fn
 and direct return `useMemo`

File: src/components/asset-amount-input/asset-amount-input.tsx [hunk 6]
```diff
@@ -110,29 +110,34 @@ export const AssetAmountInput: FC<AssetAmountInputProps> = ({
...
+      if (!isDefined(inputValueRef.current)) {
+        return undefined;
+      } else {
+        return isTokenInputType
+          ? tzToMutez(inputValueRef.current, decimals)
+          : dollarToTokenAmount(inputValueRef.current, decimals, newExchangeRate);
+      }
+    },
+    [isTokenInputType, inputValueRef.current]
+  );
```
Comment (unixvb) [inline src/components/asset-amount-input/asset-amount-input.tsx:132]: use short return

File: src/components/attention-message/attention-message.tsx [hunk 7]
```diff
@@ -8,11 +8,11 @@ import { Icon } from '../icon/icon';
 import { IconNameEnum } from '../icon/icon-name.enum';
 import { useAttentionMessageStyles } from './attention-message.styles';
 
-type AttentionMessage = {
+type AttentionMessageProps = {
```
Comment (unixvb) [inline src/components/attention-message/attention-message.tsx:11]: use `interface Props`

File: src/components/formatted-amount.tsx [hunk 8]
```diff
@@ -25,11 +25,10 @@ export const FormattedAmount: FC<Props> = ({
   symbol,
   style
 }) => {
-  const dollarAmount = amount.isZero()
-    ? amount
-    : amount.isPositive()
```
Comment (unixvb) [inline src/components/formatted-amount.tsx:30]: revert

File: src/components/password-strength-indicator/password-strength-indicator-item/password-strength-indicator-item.tsx [hunk 9]
```diff
@@ -15,7 +15,8 @@ export const PasswordStrengthIndicatorItem: FC<PasswordStrengthIndicatorItemProp
   noColor = false
 }) => {
   const styles = usePasswordStrengthIndicatorItemStyles();
-  const validationColor = isValid ? styles.adding : noColor ? undefined : styles.destructive;
```
Comment (unixvb) [inline src/components/password-strength-indicator/password-strength-indicator-item/password-strength-indicator-item.tsx:18]: revert

File: src/components/token-screen-content-container/token-screen-content-container.tsx [hunk 10]
```diff
@@ -27,22 +27,17 @@ export const TokenScreenContentContainer: FC<Props> = ({ historyComponent, infoC
...
-              <Text style={styles.delegateText}>Rewards & Redelegate</Text>
-            ) : (
-              <Text style={styles.delegateText}>
-                Delegate: <Text style={styles.apyText}>{delegationApy}% APY</Text>
-              </Text>
-            )}
+            <SelectedBaker isBakerSelected={isBakerSelected} />
           </TouchableOpacity>
         ) : (
-          <Text style={styles.headerText}>{showHistoryComponent ? 'History' : 'Info'}</Text>
```
Comment (unixvb) [inline src/components/token-screen-content-container/token-screen-content-container.tsx:45]: return

File: src/hooks/use-quick-actions.hook.spec.ts [hunk 11]
```diff
@@ -16,10 +16,10 @@ describe('useQuickActions', () => {
         userInfo: { url: '' }
       }
     ]);
+
     expect(mockDeviceEventEmitter.addListener).toBeCalled();
 
     expect(mockReactNativeQuickActions.clearShortcutItems).not.toBeCalled();
-    expect(mockDeviceEventEmitter.removeListener).not.toBeCalled();
```
Comment (unixvb) [inline src/hooks/use-quick-actions.hook.spec.ts:22]: add test for `emitter.remove`

File: src/shelter/use-shelter.hook.ts [hunk 12]
```diff
@@ -30,107 +29,11 @@ export const useShelter = () => {
...
-
-            dispatch(setIsBiometricsEnabled(true));
-            navigate(StacksEnum.MainStack);
-          }
-        })
+      getImportWalletStream(importWallet$, dispatch),
+      getCreateHdAccountStream(createHdAccount$, accounts, dispatch),
+      getCreateImportAccountStream(createImportedAccount$, accounts, dispatch, goBack),
+      getMergeStream(revealSecretKey$, revealSeedPhrase$),
+      getEnableBiometryPasswordStream(enableBiometryPassword$, dispatch, navigate)
```
Comment (unixvb) [inline src/shelter/use-shelter.hook.ts:36]: rename stream => subscription

File: src/store/wallet/wallet-epics.ts [hunk 13]
```diff
@@ -38,6 +38,13 @@ import {
   waitForOperationCompletionAction
 } from './wallet-actions';
 
+const submitLoadActions = () => [
```
Comment (unixvb) [inline src/store/wallet/wallet-epics.ts:41]: updateDataActions

File: src/utils/param-preview.utils.ts [hunk 14]
```diff
@@ -9,15 +9,106 @@ import { tzToMutez } from './tezos.util';
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
 export const getParamPreview = (opParam: ParamsWithKind): ParamPreviewInterface => {
+  const previewParamsDelegation = getParamPreviewIfOpParamDelegation(opParam);
```
Comment (unixvb) [inline src/utils/param-preview.utils.ts:12]: revert

**Review by unixvb — APPROVED** [review]

**Review by LendiHop — APPROVED** [review]

**Pull request merged**